annotation-server/src/medications/medications.service.ts

```diff
@@ -22,17 +23,100 @@ export class MedicationsService {
         private medicationRepository: Repository<Medication>,
     ) {}
 
-    async clearAllMedicationData(): Promise<void> {
-        await this.medicationRepository.delete({});
+    getAll(
+        options: FindManyOptions<Medication> = {
+            select: ['id', 'name', 'description', 'synonyms', 'drugclass'],
+        },
+    ): Promise<Medication[]> {
+        return this.medicationRepository.find(options);
+    }
+
+    getOne(options: FindOneOptions<Medication>): Promise<Medication> {
+        return this.medicationRepository.findOneOrFail(options);
+    }
+
+    getDetails(id: number): Promise<Medication> {
+        return this.getOne({
+            where: { id },
+            relations: [
+                'guidelines',
+                'guidelines.genePhenotype.phenotype',
+                'guidelines.genePhenotype.geneSymbol',
+            ],
+        }).catch(() => {
+            throw new NotFoundException('Medication could not be found!');
+        });
+    }
+
+    async findMatchingMedications(query = ''): Promise<Medication[]> {
```

Comment:
This huge SQL-statement doesn't look that nice. Wouldn't the [QueryBuilder](https://orkhan.gitbook.io/typeorm/docs/select-query-builder) of TypeORM be in alternative? There also seems to be a way to include the [Case-Statement](https://github.com/typeorm/typeorm/issues/3167#issuecomment-465339040).
I would also think that the query as it is now, is not safe against SQL injection. The QueryBuilder would solve that problem as well (I guess).

Reply:
I've changed to use prepared statements instead, which make the thing more secure in terms of sql injections. I've created a follow up issue for making this use query builder, since basti expressed his wish to do it #257